kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryptionFilter.java

```diff
@@ -94,6 +98,9 @@ private CompletionStage<ProduceRequestData> maybeEncodeProduce(ProduceRequestDat
                         });
                     }).toList();
                     return join(futures).thenApply(x -> request);
+                }).exceptionallyCompose(throwable -> {
+                   log.warn("failed to encrypt records", throwable);
```

Comment:
I needed this because I messed up the mocking and the exceptions were being silently swallowed. 

I'm not convinced that logging at warn is the entirely correct response but I think we should think about the error handling.

Reply:
I'm fine with adding this here for now. But can we add a similar `exceptionallyCompose` on the fetch side too, at the same time. So at least the "error handling" is being done the same on both paths.